serverless/serverless
# fixes #4959 allow asterix literals in variable syntax

This PR addresses issue #4959.

The previous regex used to look for candidate variables was not allowing expressions with `*`'s, this PR changes all instances of said regex 

`\\${([ ~:a-zA-Z0-9._\'",\\-\\/\\(\\)]+?)}`
 to 
`\${(([ ~:a-zA-Z0-9._'",\\-\\/\\(\\)]|\*(?=([ ~:a-zA-Z0-9._'",\\-\\/\\(\\)]+?)('|")))+?)}` which allows for `*` to be used as values in the variable syntax, only if they are surrounded by `'` or `"`


## How can we verify it:
1. Observe that a function deployed using *code from this patch* will have the correct STAGE_VAR (`*.mystage.com`)
2. Observe that a function deployed using *code deployed without this patch* will have the incorrect STAGE_VAR (`${opt:stage, '*.mystage.com'}`)

```yaml
service: test-sls-changes

custom:
  stage: ${opt:stage, '*.mystage.com'}

provider:
  name: aws
  runtime: nodejs6.10
  environment:
    STAGE_VAR: ${self:custom.stage}

functions:
  hello:
    handler: handler.hello
```

## Todos:

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
